components/atom/backToTop/src/index.js

```diff
@@ -0,0 +1,144 @@
+import React, {PureComponent} from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+import {getTarget} from './utils'
+
+const BASE_CLASS = 'sui-AtomBackToTop'
+const CLASS_ICON = 'sui-AtomBackToTop-icon'
+const CLASS_TEXT = 'sui-AtomBackToTop-text'
+const CLASS_SHOW = 'sui-AtomBackToTop--show'
+const CLASS_HIDE = 'sui-AtomBackToTop--hide'
+const CLASS_HOVER = 'sui-AtomBackToTop--hover'
+const CLASS_READY = 'sui-AtomBackToTop--ready'
+
+const STYLES = {
+  DARK: 'dark',
+  LIGHT: 'light'
+}
+
+class AtomBackToTop extends PureComponent {
+  state = {
+    show: null,
+    hover: false
```

Comment:
Couldn't this be done via css directly with :hover? Looks like you are just adding a css class

Reply:
I found 2 problems when using the native css `hover`:
- The `AtomBackToTop` doesn't disappear when the scroll crosses the half height if the pointer is still over the button (a common behaviour)
- On mobile The `AtomBackToTop` doesn't disappear even when the scroll reaches the top (it needs a click outisde the button to get the `unhover` event )

By Managing this through JS I got more control over this and I could solve these issues

You can see these problems when implementing this w/ the native CSS `hover` here → https://sui-components-feature-atom-back-to-top-hover-css.now.sh/workbench/atom/backToTop/demo